src/components/ArticlesMainCard/ArticlesMainCard.jsx

```diff
@@ -1,30 +1,70 @@
-import { Link } from 'react-router-dom';
-import voteIcon from '../../assets/vote-icon.svg';
+import { Link } from "react-router-dom";
+import voteIcon from "../../assets/vote-icon.svg";
+import voteDownIcon from "../../assets/vote-down-icon.svg";
+import { patchVoteInArticle } from "../../api";
+
+export default function ArticlesMainCard({ articles, setArticles }) {
+  function handleClickVote(article_id, increment) {
+    setArticles((prevArticles) =>
+      prevArticles.map((article) => {
+        if (article.article_id === article_id) {
+          const newVoteCount = article.votes + increment;
+          if (increment > 0 || (increment < 0 && article.votes > 0)) {
+            patchVoteInArticle(article_id, increment).then(() => {
+              setArticles((updatedArticles) =>
```

Comment:
You're calling setArticles within itself here which could cause some weird render issues because you're setting it twice and its hard to tell exactly what is being set and when. 

I'd suggest breaking this code apart a little more and thinking of it a little more synchronously.

It is quite low stakes as to whether the vote goes through and so they don't really rely on each other because we're doing optimistic rendering

```js
setArticles(// function to update the vote count in react)
patchVoteInArticle(// function to send the api request)
```